apis/v1alpha2/httproute_types.go

```diff
@@ -659,29 +683,32 @@ type HTTPRequestRedirect struct {
 	// +optional
 	// +kubebuilder:validation:Enum=HTTP;HTTPS
 	Protocol *string `json:"protocol,omitempty"`
+
 	// Hostname is the hostname to be used in the value of the `Location`
 	// header in the response.
 	// When empty, the hostname of the request is used.
 	//
 	// Support: Core
 	//
 	// +optional
-	Hostname *string `json:"hostname,omitempty"`
+	Hostname *Hostname `json:"hostname,omitempty"`
+
 	// Port is the port to be used in the value of the `Location`
 	// header in the response.
 	// When empty, port (if specified) of the request is used.
 	//
 	// Support: Extended
 	//
 	// +optional
-	Port *int `json:"port,omitempty"`
+	Port *PortNumber `json:"port,omitempty"`
+
 	// StatusCode is the HTTP status code to be used in response.
 	//
 	// Support: Core
 	//
 	// +optional
 	// +kubebuilder:default=302
-	// +kubebuilder:validation=301;302
+	// +kubebuilder:validation:Enum=301;302
```

Comment:
I feel like we should just include all the valid redirects from here:

https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#redirection_messages

Reply:
Agree that we should add this, but didn't want to expand the scope of this PR so added https://github.com/kubernetes-sigs/gateway-api/issues/793 to track this.

Reply:
There was a conformance concern for not adding them in the first place.